### PR TITLE
Add support for "Asia/Qostanay" IANA time zone.

### PR DESCRIPTION
**Pull request by gregakespret**
## Problem statement
Getting this error when running tests:

```
io.prestosql.spi.type.TimeZoneNotSupportedException: Time zone not supported: Asia/Qostanay
    at io.prestosql.spi.type.TimeZoneKey.getTimeZoneKey(TimeZoneKey.java:132)
    at io.prestosql.util.TestTimeZoneUtils.test(TestTimeZoneUtils.java:70)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
    at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
    at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
    at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
    at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
    at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:748)
```

```
$ java -version
java version "1.8.0_221"
Java(TM) SE Runtime Environment (build 1.8.0_221-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.221-b11, mixed mode)
```

## Root cause

Timezone **Asia/Qostanay** was introduced in JRE Update Releases 12.0.2, 11.0.4, 8u221 and 7u231 with Timezone Tzdata version tzdata2018h.

- https://www.oracle.com/technetwork/java/javase/tzdata-versions-138805.html
- http://mm.icann.org/pipermail/tz-announce/2018-December/000053.html

### CODE REVIEW

**Comment by electrum** [comment]
@cla-bot check

**Comment by gregakespret** [comment]
@electrum Sure. I'm getting a build error however, no idea what to check.

```[ERROR] An error occurred while instantiating class io.prestosql.type.TestTimeLegacy: null
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] 
[ERROR] An error occurred while instantiating class io.prestosql.type.TestTimeLegacy: null
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:658)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:533)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:278)
[ERROR] 	at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:244)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1194)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1022)
[ERROR] 	at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:868)
[ERROR] 	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)
[ERROR] 	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
[ERROR] 	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
[ERROR] 	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)
[ERROR] 	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)
[ERROR] 	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)
[ERROR] 	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)
[ERROR] 	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)
[ERROR] 	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
[ERROR] 	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.lang.reflect.Method.invoke(Method.java:498)
[ERROR] 	at org.apache.maven.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:39)
[ERROR] 	at org.apache.maven.wrapper.WrapperExecutor.execute(WrapperExecutor.java:122)
[ERROR] 	at org.apache.maven.wrapper.MavenWrapperMain.main(MavenWrapperMain.java:50)
[ERROR] -> [Help 1]```

**Comment by findepi** [comment]
> Sure. I'm getting a build error however, no idea what to check.

@gregakespret maven Surefire trims certain exceptions during test initialization.
I run the `TestTimeLegacy` test on your change in IDE and got:
```
org.testng.TestNGException: 
An error occurred while instantiating class io.prestosql.type.TestTimeLegacy: null
	at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:375)
	at org.testng.internal.ClassHelper.createInstance(ClassHelper.java:275)
	at org.testng.internal.ClassImpl.getDefaultInstance(ClassImpl.java:126)
	at org.testng.internal.ClassImpl.getInstances(ClassImpl.java:191)
	at org.testng.TestClass.getInstances(TestClass.java:100)
	at org.testng.TestClass.initTestClassesAndInstances(TestClass.java:86)
	at org.testng.TestClass.init(TestClass.java:78)
	at org.testng.TestClass.<init>(TestClass.java:41)
	at org.testng.TestRunner.initMethods(TestRunner.java:425)
	at org.testng.TestRunner.init(TestRunner.java:252)
	at org.testng.TestRunner.init(TestRunner.java:222)
	at org.testng.TestRunner.<init>(TestRunner.java:163)
	at org.testng.SuiteRunner$DefaultTestRunnerFactory.newTestRunner(SuiteRunner.java:585)
	at org.testng.SuiteRunner.init(SuiteRunner.java:189)
	at org.testng.SuiteRunner.<init>(SuiteRunner.java:136)
	at org.testng.TestNG.createSuiteRunner(TestNG.java:1375)
	at org.testng.TestNG.createSuiteRunners(TestNG.java:1355)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1209)
	at org.testng.TestNG.runSuites(TestNG.java:1133)
	at org.testng.TestNG.run(TestNG.java:1104)
	at org.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:73)
	at org.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:123)
Caused by: java.lang.ExceptionInInitializerError
	at io.prestosql.type.TestTimeBase.<clinit>(TestTimeBase.java:46)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.testng.internal.ObjectFactoryImpl.newInstance(ObjectFactoryImpl.java:29)
	at org.testng.internal.ClassHelper.createInstance1(ClassHelper.java:363)
	... 21 more
Caused by: java.lang.IllegalArgumentException: The datetime zone id 'Asia/Qostanay' is not recognised
	at org.joda.time.DateTimeZone.forID(DateTimeZone.java:247)
	at io.prestosql.util.DateTimeZoneIndex.<clinit>(DateTimeZoneIndex.java:45)
	... 28 more
```

(This is even though Joda is updated to 2.10.3.)

**Comment by gregakespret** [comment]
Aha, that's more informative. It looks like Joda fails (`org.joda.time.DateTimeZone.forID(DateTimeZone.java:247)`); not sure why.

Test passes locally for me in the IDE, I'm running it with Java 8u221 however. Don't see how that should matter though.

**Comment by findepi** [comment]
> Test passes locally for me in the IDE, I'm running it with Java 8u221 however. Don't see how that should matter though.

Ah, right. It fails for me because the tests still use the bridge and I've 8u172 installed.

Therefore I propose let's merge https://github.com/prestosql/presto/pull/1222 first. (There, rebase to exclude commits that are part of this PR).

However, there still remains a problem what to do with Travis. Currently it runs `openjdk version "1.8.0_191"`. I don't have answer yet.

**Comment by gregakespret** [comment]
> Ah, right. It fails for me because the tests still use the bridge and I've 8u172 installed.

Good catch!

> Therefore I propose let's merge #1222 first. (There, rebase to exclude commits that are part of this PR).

OK.

**Comment by gregakespret** [comment]
@findepi I rebased this on top of latest master now that #1222 was merged. The tests are still failing though (2x in a row), so I'm not sure if the tests are flaky or something really is wrong. Could you take a look?

File: presto-main/src/test/java/io/prestosql/util/TestTimeZoneUtils.java [hunk 1]
```diff
@@ -35,7 +35,6 @@ public void test()
     {
         TimeZoneKey.getTimeZoneKey("GMT-13:00");
 
-        TreeSet<String> jodaZones = new TreeSet<>(DateTimeZone.getAvailableIDs());
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/util/TestTimeZoneUtils.java]: This is not related to adding a new zone, it's an unrelated cleanup.
Normally you'd extract this to a separate commit (within this PR), but there are other things to clean up in this class, so I put up https://github.com/prestosql/presto/pull/1249.
Thus, you can drop this change from here.

**Comment by findepi** [comment]
@gregakespret i checked Travis and there was one build failure due to Docker registry IO error. I retriggered, let's see if it goes OK now.

File: presto-main/src/test/java/io/prestosql/util/TestTimeZoneUtils.java (hunk 1 above)
Comment (gregakespret) [reply presto-main/src/test/java/io/prestosql/util/TestTimeZoneUtils.java]: @findepi Sure, removed from this PR. Originally, forgot to delete that line in https://github.com/prestosql/presto/pull/1222

**Comment by findepi** [comment]
Merged as 7c6c31610a2324e08d9bacfa12747e3e68e28827, thanks!

(I adjusted commit message slightly.)
